airflow/models/log.py

```diff
@@ -55,7 +55,9 @@ def __init__(self, event, task_instance, owner=None, extra=None, **kwargs):
             self.dag_id = task_instance.dag_id
             self.task_id = task_instance.task_id
             self.execution_date = task_instance.execution_date
-            task_owner = task_instance.task.owner
+            if owner is None:
+                # sometimes we want to use this when task is not set for task_instance
```

Comment:
This comment reads a bit weird. What is it saying? And can you remind me when owner is set vs not set? I remember we looked at this together a while back.

Reply:
will clarify this. tldr is that the task is only set in the init method of the TaskInstance class so when the task_instance passed in is retrieved via sqlalchemy query, the task property isnt set and so the owner isn't discoverable that way. also we dont necessarily want the dag owner, but the person who issued the request